### DISCUSSION TITLE
What collection changes are safe?

### DISCUSSION THREAD

**Discussion by chriselster** [isar/isar#1217] [Help] [answered]
I have searched through the documentation and discussions, but I have only been able to find fragmented answers. I am using Isar for a fairly large application and I would like to know which changes to my collections will be automatically migrated when I run the code generator to ensure that users do not lose their data.

Thanks in advance!

**Comment by simc** [comment, marked as answer]
There is no disallowed modification. Just make sure you use `3.1.0` as there has been a problem with collection migration before.
